google/cloud/storage/blob.py

```diff
@@ -1822,6 +1911,16 @@ def _do_resumable_upload(
             Can also be passed as a tuple (connect_timeout, read_timeout).
             See :meth:`requests.Session.request` documentation for details.
 
+        :type checksum: str
+        :param checksum:
+            (Optional) The type of checksum to compute to verify
+            the integrity of the object. After the upload is complete, the
+            server-computed checksum of the resulting object will be checked
+            and google.resumable_media.common.DataCorruption will be raised on
+            a mismatch. On a failure, the client will attempt to delete the
+            corrupted file from the remote host automatically. Supported values
+            are "md5", "crc32c" and None. The default is None.
```

Comment:
`On a failure, the client will attempt to delete the corrupted file from the remote host automatically` 

Suggestion:
`On a validation failure, the client will attempt to delete the uploaded object automatically.`

Reply:
Done